### PR TITLE
chore(merge-variables): allow nested merge variables

### PR DESCRIPTION
**Pull request by ajpawlicki**
## What
Allows nested merge variables on resource creation

## Why
Advanced templating syntax

### CODE REVIEW

File: tests/Lob/Tests/Resource/ChecksTest.php [hunk 1]
```diff
@@ -46,6 +66,15 @@ public function testCreate()
         $this->assertTrue(array_key_exists('id', $check));
     }
 
+    public function testCreateWithMergeVariableBoolean()
+    {
+        $check = $this->lob->checks()->create($this->mergeVariableBooleanCheckParams);
+
+        $this->assertTrue(is_array($check));
+        $this->assertTrue(array_key_exists('id', $check));
+        $this->assertTrue($check['merge_variables']['is_awesome']);
```
Comment (mpiercy827) [inline tests/Lob/Tests/Resource/ChecksTest.php:75]: Do we want to have an assertion for the attachment HTML?

File: tests/Lob/Tests/Resource/LettersTest.php [hunk 2]
```diff
@@ -51,6 +63,15 @@ public function testColorCreate()
         $this->assertTrue(array_key_exists('id', $letter));
     }
 
+    public function testCreateWithMergeVariableObject()
+    {
+        $letter = $this->lob->letters()->create($this->mergeVariableObjectLetterParams);
+
+        $this->assertTrue(is_array($letter));
+        $this->assertTrue(array_key_exists('id', $letter));
+        $this->assertTrue($letter['merge_variables']['data']['name'] === 'Kobe');
```
Comment (mpiercy827) [inline tests/Lob/Tests/Resource/LettersTest.php:72]: Same as above: do we need to check the file HTML?

File: tests/Lob/Tests/Resource/PostcardsTest.php [hunk 3]
```diff
@@ -35,6 +52,16 @@ public function testCreate()
...
     }
 
+    public function testCreateWithMergeVariableList()
+    {
+        $postcard = $this->lob->postcards()->create($this->mergeVariableListPostcardParams);
+
+        $this->assertTrue(is_array($postcard));
+        $this->assertTrue(array_key_exists('id', $postcard));
+        $this->assertTrue($postcard['merge_variables']['list'][0]['name'] === 'Larissa');
+        $this->assertTrue($postcard['merge_variables']['list'][1]['name'] === 'Larry');
```
Comment (mpiercy827) [inline tests/Lob/Tests/Resource/PostcardsTest.php:62]: same question here.

File: tests/Lob/Tests/Resource/ChecksTest.php (hunk 1 above)
Comment (ajpawlicki) [reply tests/Lob/Tests/Resource/ChecksTest.php:75]: Spoke over slack. No need to do this because the response doesn't contain inline html

**Pull request merged**